GeventConnection may now work on Windows

We should check

I'd call this a big NOPE:

```
======================================================================
FAIL: test_with_gevent (acceptance.async_adapter_tests.TestZ_PublishAndGet)
Publish a message and get it (GeventConnection)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\a\pika\pika\tests\wrappers\threaded_test_wrapper.py", line 59, in run_in_thread_with_timeout_wrapper
    return runner.kick_off()
  File "D:\a\pika\pika\tests\wrappers\threaded_test_wrapper.py", line 129, in kick_off
    raise AssertionError(self._exc_info_to_str(self._exc_info))
AssertionError: Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.10.10\x64\lib\site-packages\gevent\libuv\loop.py", line 669, in io
    io_watcher = io_watchers[fd]
KeyError: 81

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\a\pika\pika\tests\wrappers\threaded_test_wrapper.py", line 145, in _thread_entry
    self._fun_result = self._fun()
  File "D:\a\pika\pika\tests\base\async_test_base.py", line 364, in test_with_gevent
    self.start(GeventConnection, ioloop_factory)
  File "D:\a\pika\pika\tests\base\async_test_base.py", line 262, in start
    super(BoundQueueTestCase, self).start(adapter, ioloop_factory)
  File "D:\a\pika\pika\tests\base\async_test_base.py", line 158, in start
    custom_ioloop=ioloop_factory())
  File "D:\a\pika\pika\tests\base\async_test_base.py", line 362, in ioloop_factory
    return _GeventSelectorIOLoop(gevent.get_hub())
  File "D:\a\pika\pika\pika\adapters\gevent_connection.py", line 198, in __init__
    self.add_handler(self._callback_queue.fd, run_callback_in_main_thread,
  File "D:\a\pika\pika\pika\adapters\gevent_connection.py", line 281, in add_handler
    io_watcher = self._hub.loop.io(fd, events)
  File "C:\hostedtoolcache\windows\Python\3.10.10\x64\lib\site-packages\gevent\libuv\loop.py", line 675, in io
    io_watcher = self._watchers.io(self, fd, 0)
  File "C:\hostedtoolcache\windows\Python\3.10.10\x64\lib\site-packages\gevent\libuv\watcher.py", line 252, in __init__
    super(io, self).__init__(loop, fd, events, ref=ref, priority=priority, _args=(fd,))
  File "C:\hostedtoolcache\windows\Python\3.10.10\x64\lib\site-packages\gevent\_ffi\watcher.py", line 477, in __init__
    super(IoMixin, self).__init__(loop, ref=ref, priority=priority,
  File "C:\hostedtoolcache\windows\Python\3.10.10\x64\lib\site-packages\gevent\_ffi\watcher.py", line 224, in __init__
    self._watcher_full_init()
  File "C:\hostedtoolcache\windows\Python\3.10.10\x64\lib\site-packages\gevent\_ffi\watcher.py", line 238, in _watcher_full_init
    self._watcher_ffi_init(args)
  File "C:\hostedtoolcache\windows\Python\3.10.10\x64\lib\site-packages\gevent\libuv\watcher.py", line 163, in _watcher_ffi_init
    return self._watcher_init(self.loop.ptr,
  File "C:\hostedtoolcache\windows\Python\3.10.10\x64\lib\site-packages\gevent\libuv\watcher.py", line 82, in wrap
    raise UVFuncallError(
gevent.libuv.watcher.UVFuncallError: ENOTSOCK socket operation on non-socket Args: (<cdata 'struct uv_loop_s *' 0x000002955E58C3F0>, <cdata 'struct uv_poll_s *' owning 416 bytes>, 81) KWARGS: {}
```

--- a/testdata/versions.json
+++ b/testdata/versions.json
@@ -1,4 +1,4 @@
 {
-  "erlang": "25.1.2",
-  "rabbitmq": "3.11.3"
+  "erlang": "25.2.3",
+  "rabbitmq": "3.11.9"
 }

--- a/tests/base/async_test_base.py
+++ b/tests/base/async_test_base.py
@@ -32,9 +32,7 @@ from tests.wrappers.threaded_test_wrapper import create_run_in_thread_decorator
 # protected-access
 # pylint: disable=W0212
 
-
 TEST_TIMEOUT = 15
-
 
 # Decorator for running our tests in threads with timeout
 # NOTE: we give it a little more time to give our I/O loop-based timeout logic
@@ -53,6 +51,7 @@ def make_stop_on_error_with_self(the_self=None):
         exception otherwise.
 
     """
+
     def stop_on_error_with_self_decorator(fun):
 
         @functools.wraps(fun)
@@ -158,16 +157,17 @@ class AsyncTestCase(unittest.TestCase):
                                        self.on_closed,
                                        custom_ioloop=ioloop_factory())
         try:
-            self.timeout = self.connection._adapter_call_later(self.TIMEOUT,
-                                                               self.on_timeout)
+            self.timeout = self.connection._adapter_call_later(
+                self.TIMEOUT, self.on_timeout)
             self._run_ioloop()
 
             self.assertFalse(self._timed_out)
             self.assertIsNone(self._conn_open_error)
             # Catch unexpected loss of connection
-            self.assertTrue(self._public_stop_requested,
-                            'Unexpected end of test; connection close reason: '
-                            '{!r}'.format(self._conn_closed_reason))
+            self.assertTrue(
+                self._public_stop_requested,
+                'Unexpected end of test; connection close reason: '
+                '{!r}'.format(self._conn_closed_reason))
             if self._public_stop_error_in is not None:
                 raise self._public_stop_error_in  # pylint: disable=E0702
         finally:
@@ -198,7 +198,7 @@ class AsyncTestCase(unittest.TestCase):
         self.logger.info('Stopping test')
         self._public_stop_requested = True
         if self.connection.is_open:
-            self.connection.close() # NOTE: on_closed() will stop the ioloop
+            self.connection.close()  # NOTE: on_closed() will stop the ioloop
         elif self.connection.is_closed:
             self.logger.info(
                 'Connection already closed, so just stopping ioloop')
@@ -244,8 +244,8 @@ class AsyncTestCase(unittest.TestCase):
 
     def on_timeout(self):
         """called when stuck waiting for connection to close"""
-        self.logger.error('%s timed out; on_timeout called at %s',
-                          self, datetime.datetime.utcnow())
+        self.logger.error('%s timed out; on_timeout called at %s', self,
+                          datetime.datetime.utcnow())
         self.timeout = None  # the dispatcher should have removed it
         self._timed_out = True
         # initiate cleanup
@@ -279,7 +279,8 @@ class BoundQueueTestCase(AsyncTestCase):
                                    callback=self.on_queue_declared)
 
     def on_queue_declared(self, frame):  # pylint: disable=W0613
-        self.channel.queue_bind(self.queue, self.exchange,
+        self.channel.queue_bind(self.queue,
+                                self.exchange,
                                 self.routing_key,
                                 callback=self.on_ready)
 
@@ -293,6 +294,7 @@ class BoundQueueTestCase(AsyncTestCase):
 # from the AsyncAdapters class below. This allows you to avoid duplicating the
 # test methods for each adapter in each test class.
 #
+
 
 class AsyncAdapters(object):
 
@@ -324,9 +326,8 @@ class AsyncAdapters(object):
         with mock.patch.multiple(select_connection, SELECT_TYPE='select'):
             self.start(adapters.SelectConnection, select_connection.IOLoop)
 
-    @unittest.skipIf(
-        not hasattr(select, 'poll') or
-        not hasattr(select.poll(), 'modify'), "poll not supported")  # pylint: disable=E1101
+    @unittest.skipIf(not hasattr(select, 'poll') or
+                     not hasattr(select.poll(), 'modify'), "poll not supported")  # pylint: disable=E1101
     @run_test_in_thread_with_timeout
     def test_with_select_poll(self):
         """SelectConnection:poll"""

--- a/tests/unit/io_services_test_stubs_test.py
+++ b/tests/unit/io_services_test_stubs_test.py
@@ -1,7 +1,8 @@
 """
 Test for io_services_test_stubs.py
-
 """
+
+import pika.compat
 
 try:
     import asyncio
@@ -19,14 +20,12 @@ from pika.adapters import select_connection
 
 from tests.stubs.io_services_test_stubs import IOServicesTestStubs
 
-
 # Suppress invalid-name, since our test names are descriptive and quite long
 # pylint: disable=C0103
 
 # Suppress missing-docstring to allow test method names to be printed by our the
 # test runner
 # pylint: disable=C0111
-
 
 # Tornado does some magic that substitutes the class dynamically
 _TORNADO_IO_LOOP = tornado.ioloop.IOLoop()
@@ -40,14 +39,13 @@ _SUPPORTED_LOOP_CLASSES = {
 }
 
 if asyncio is not None:
-    if sys.platform == 'win32':
+    if pika.compat.ON_WINDOWS:
         asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
     _SUPPORTED_LOOP_CLASSES.add(asyncio.get_event_loop().__class__)
 
 
 class TestStartCalledFromOtherThreadAndWithVaryingNativeLoops(
-        unittest.TestCase,
-        IOServicesTestStubs):
+        unittest.TestCase, IOServicesTestStubs):
 
     _native_loop_classes = None
 
@@ -58,7 +56,7 @@ class TestStartCalledFromOtherThreadAndWithVaryingNativeLoops(
     # AssertionError: Expected these 3 native I/O loop classes from IOServicesTestStubs:
     # {<class 'asyncio.windows_events.ProactorEventLoop'>, <class 'tornado.platform.asyncio.AsyncIOLoop'>, <class 'pika.adapters.select_connection.IOLoop'>}
     # but got these 3:
-    # {<class 'asyncio.windows_events._WindowsSelectorEventLoop'>, <class 'tornado.platform.asyncio.AsyncIOLoop'>, <class 'pika.adapters.select_connection.IOLoop'>} 
+    # {<class 'asyncio.windows_events._WindowsSelectorEventLoop'>, <class 'tornado.platform.asyncio.AsyncIOLoop'>, <class 'pika.adapters.select_connection.IOLoop'>}
     @classmethod
     def tearDownClass(cls):
         # Now check against what was made available to us by
@@ -67,10 +65,8 @@ class TestStartCalledFromOtherThreadAndWithVaryingNativeLoops(
             raise AssertionError(
                 'Expected these {} native I/O loop classes from '
                 'IOServicesTestStubs: {!r}, but got these {}: {!r}'.format(
-                    len(_SUPPORTED_LOOP_CLASSES),
-                    _SUPPORTED_LOOP_CLASSES,
-                    len(cls._native_loop_classes),
-                    cls._native_loop_classes))
+                    len(_SUPPORTED_LOOP_CLASSES), _SUPPORTED_LOOP_CLASSES,
+                    len(cls._native_loop_classes), cls._native_loop_classes))
 
     def setUp(self):
         self._runner_thread_id = threading.current_thread().ident

